plugins/module_utils/adguardhome/api.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.module_utils.basic import (
+    missing_required_lib
+)
+import traceback
+
+REQUESTS_IMP_ERR = None
+try:
+    import requests
+    HAS_REQUESTS = True
+except Exception:
+    REQUESTS_IMP_ERR = traceback.format_exc()
+    HAS_REQUESTS = False
```

Comment:
Is there a reason why you use requests instead of `ansible.module_utils.urls`? The latter should always be used unless there is a very important reason not to (for example if you're using a third-party library that already uses requests, or need some requests functionality that's not provided by ansible-core and would be hard to implement).